src/main/java/io/gravitee/resource/ai_model/api/AiModelResource.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.resource.ai_model.api;
+
+import io.gravitee.resource.api.AbstractConfigurableResource;
+import io.gravitee.resource.api.ResourceConfiguration;
+import io.reactivex.rxjava3.core.Maybe;
+
+public abstract class AiModelResource<C extends ResourceConfiguration> extends AbstractConfigurableResource<C> {
+
+    public abstract Maybe<ClassifierResults> invokeModel(String input);
```

Comment:
Should we rename the abstract class `AiTextClassificationModelResource` instead because we are returning a ClassifierResult that may not relevant for another kind of model?